quickfixj-core/src/main/java/quickfix/DefaultSessionSchedule.java

```diff
@@ -300,9 +307,16 @@ public boolean isSessionTime() {
         if(isNonStopSession()) {
             return true;
         }
-        Calendar now = SystemTime.getUtcCalendar();
-        TimeInterval interval = theMostRecentIntervalBefore(now);
-        return interval.isContainingTime(now);
+        Calendar now = threadLocalCalendar.get();
+        now.setTimeInMillis(SystemTime.currentTimeMillis());
+        TimeInterval mostRecentInterval = threadLocalRecentTimeInterval.get();
+        if (mostRecentInterval != null && mostRecentInterval.isContainingTime(now)) {
```

Comment:
This optimisation should mean that during session time we don't have to keep reconstructing time intervals and instead just re-use one. Constructing a `TimeInterval` involves a bunch of throwaway Calendar usage.